compiler-rt/src/crt0/memory/atomicrmw/atomicrmw_add.cairo

```diff
@@ -0,0 +1,190 @@
+use crate::crt0::allocator::{AllocatorState, Address};
+use crate::crt0::memory::{load::load, store::store};
+use core::traits::{BitAnd, BitOr};
+use core::num::traits::{BitSize, OverflowingMul, WrappingAdd};
+use crate::crt0::memory::get_allocator;
+use crate::integer::{u24::u24, u40::u40, u48::u48};
+
+/// Atomically modify memory. Load a value from memory, add the given value to it, and store the
+/// result into the memory. The value loaded from the memory is returned. The number of loaded and
+/// stored bytes is equal to the size of T.
+///
+/// Since the Cairo VM is single-threaded, this operation does not need to be explicitly atomic.
+/// The load-modify-store sequence will not be interrupted.
```

Comment:
I think we should be explicit about the fact that it **cannot** be interrupted, not just that it "will not" be interrupted.